runtime-modules/project-token/src/lib.rs

```diff
@@ -0,0 +1,611 @@
+                // if no more account for token -> deissue
+                if AccountInfoByTokenAndAccount::<T>::iter_prefix(token_id)
+                    .next()
+                    .is_none()
+                {
+                    Self::do_deissue_token(token_id)
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    // /// Transfer `amount` from `src` account to `dst`
+    // /// Preconditions:
+    // /// - `token_id` must exists
+    // /// - `src` must exists
+    // /// - `src` free balance must be greater than or equal to `amount`
+    // /// - `dst` must exists
+    // ///
+    // /// Postconditions:
+    // /// - free balance of `src` is decreased by `amount or set to zero if below existential
+    // ///   deposit`
+    // /// - free balance of `dst` is increased by `amount`
+    // /// if `amount` is zero it is equivalent to a no-op
+    // fn transfer(
+    //     token_id: T::TokenId,
+    //     src: T::AccountId,
+    //     dst: T::AccountId,
+    //     amount: T::Balance,
+    // ) -> DispatchResult {
+    //     if amount.is_zero() {
+    //         return Ok(());
+    //     }
+
+    //     // tranfer preconditions
+    //     let outputs = [(SimpleLocation::<T::AccountId>::new(dst.clone()), amount)];
+    //     let (decrease_operation, _) = Self::ensure_can_transfer(token_id, &src, &outputs)?;
+
+    //     // == MUTATION SAFE ==
+
+    //     Self::do_transfer(token_id, &src, &outputs, decrease_operation);
+
+    //     Self::deposit_event(RawEvent::TokenAmountTransferred(token_id, src, dst, amount));
+    //     Ok(())
+    // }
+
+    /// Issue token with specified characteristics
+    /// Preconditions:
+    /// -
+    ///
+    /// Postconditions:
+    /// - token with specified characteristics is added to storage state
+    /// - `NextTokenId` increased by 1
+    fn issue_token(issuance_parameters: TokenIssuanceParametersOf<T>) -> DispatchResult {
+        let token_data = issuance_parameters.try_build::<T>()?;
+
+        // == MUTATION SAFE ==
+
+        let token_id = Self::next_token_id();
+        TokenInfoById::<T>::insert(token_id, token_data);
+        NextTokenId::<T>::put(token_id.saturating_add(T::TokenId::one()));
+
+        Ok(())
+    }
+
+    /// Remove token data from storage
+    /// Preconditions:
+    /// - `token_id` must exists
+    ///
+    /// Postconditions:
+    /// - token data @ `token_Id` removed from storage
+    /// - all account data for `token_Id` removed
+    fn deissue_token(token_id: T::TokenId) -> DispatchResult {
+        Self::ensure_token_exists(token_id).map(|_| ())?;
+
+        // == MUTATION SAFE ==
+
+        Self::do_deissue_token(token_id);
+        Ok(())
+    }
+
+    /// Retrieve usable (free) balance for token and account
+    /// Preconditions:
+    /// - `token_id` must be a valid token identifier
+    /// - `who` account id must exist for token
+    fn balance(token_id: T::TokenId, who: T::AccountId) -> Result<T::Balance, DispatchError> {
```

Comment:
Why do we have this?

Reply:
removed - dead code

Reply:
in b2726deded2bf71154e0e8e5e8ad01c4fbaaa257